markdown/events/2021/bytesize-12-template-sync.md

```diff
@@ -38,3 +38,94 @@ The talk will be presented on Zoom and live-streamed on YouTube:
+
+[5:39](https://youtu.be/-CZKoo5Y_08?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=337) So hopefully that made sense.
+
+[5:44](https://youtu.be/-CZKoo5Y_08?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=344) Finally, we automate this process, so that when we do a new release of the tools repository, all these changes can happen automatically and the pull requests are automatically opened.
+
+[5:57](https://youtu.be/-CZKoo5Y_08?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=357) So what does this synchronisation process do step by step? It’s pretty simple. First of all, it gets a copy of your pipeline and checks out the special `TEMPLATE` branch and deletes everything in that pipeline. Then it creates a new pipeline from scratch using `nf core create`. This time of course what you’re using is the latest release of the nf-core tools package, and so it uses the latest version of a `TEMPLATE`. So this new version now has all the modifications that we’ve made since the last release. Those changes, if there are any, are committed into a `TEMPLATE` branch (we’ve recently added a feature where we make a special branch and I’ll go into why). Then we open a `pull request` with these changes to the dev branch, and Git takes care of the rest. You’ve been doing thousands of `commit`s on the `dev` and `TEMPLATE` branches, and a whole lot of other things, but Git picks what’s relevant. So then you review that `pull request`and have the latest changes.
+
+[7:33](https://youtu.be/-CZKoo5Y_08?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=453) We’ve got something special for you today, which is the `nf-core/tools` here. We’ve been working hard on a new release, and in fact it’s all prepped and ready to go.
+
+[7:46](https://youtu.be/-CZKoo5Y_08?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=466) Here we are on the release page. I’ve just written the changelog and we’ve merged the master, all tests are passed, so I’m going to hit release now. We will see if the synchronisation process works (automated sync doesn’t work completely flawlessly all the way and will mostly fail).
+
+[8:09](https://youtu.be/-CZKoo5Y_08?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=489) So I hit publish, and there you go, tools is released!
+
+[8:13](https://youtu.be/-CZKoo5Y_08?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=494) Now if I go to the Github access page, you can see lots of stuff is automatically happening here because of the release. The one that we are interested in is called `sync template`.
```

Comment:
```suggestion
[8:13](https://youtu.be/-CZKoo5Y_08?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=494) Now if I go to the Github actions page, you can see lots of stuff is automatically happening here because of the release. The one that we are interested in is called `sync template`.
```